### PR TITLE
Fix #378: allow setting of dice vendor/secret keys for rsiviz and devbox

### LINKED ISSUE #378: rsiviz and devbox allow setting of dice keys

### PR DESCRIPTION
**Pull request by e-carlin**

### CODE REVIEW

File: rsconf/component/rsiviz.py [hunk 1]
```diff
@@ -57,12 +57,15 @@ def internal_build_write(self):
...
+        if "dice" in z:
+            e.dice = z.dice
         systemd.docker_unit_enable(
             self,
             jc,
             env=self.python_service_env(
-                PKDict(rsiviz=z, pykern=jc.pykern),
-                exclude_re=r"^rsiviz(?:__|_docker_image|_url_secret)",
+                e,
+                exclude_re=r"^rsiviz(?:__|_docker_image|_url_secret|_dice_)",
```
Comment (robnagler) [inline rsconf/component/rsiviz.py:68]: Just a comment: Perhaps it should be include_re ?

File: tests/pkcli/build_data/1.out/srv/host/v9.radia.run/srv/rsiviz/cmd [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
```
Comment (robnagler) [inline tests/pkcli/build_data/1.out/srv/host/v9.radia.run/srv/rsiviz/cmd:1]: I don't find this file to be useful to include in test data.

File: tests/pkcli/build_data/1.out/srv/host/v9.radia.run/rsiviz.sh [hunk 3]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
```
Comment (robnagler) [inline tests/pkcli/build_data/1.out/srv/host/v9.radia.run/rsiviz.sh:1]: Don't include this file.

File: tests/pkcli/build_data/1.out/srv/host/v9.radia.run/srv/rsiviz/remove [hunk 4]
```diff
@@ -0,0 +1,2 @@
+#!/bin/bash
```
Comment (robnagler) [inline tests/pkcli/build_data/1.out/srv/host/v9.radia.run/srv/rsiviz/remove:1]: don't include this

File: tests/pkcli/build_data/1.out/srv/host/v9.radia.run/srv/rsiviz/start [hunk 5]
```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
```
Comment (robnagler) [inline tests/pkcli/build_data/1.out/srv/host/v9.radia.run/srv/rsiviz/start:1]: don't include this

File: tests/pkcli/build_data/1.out/srv/host/v9.radia.run/srv/rsiviz/stop [hunk 6]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
```
Comment (robnagler) [inline tests/pkcli/build_data/1.out/srv/host/v9.radia.run/srv/rsiviz/stop:1]: don't include this

**Review by robnagler — CHANGES_REQUESTED** [review]

File: tests/pkcli/build_data/1.out/srv/host/v9.radia.run/srv/rsiviz/stop (hunk 6 above)
Comment (e-carlin) [reply tests/pkcli/build_data/1.out/srv/host/v9.radia.run/srv/rsiviz/stop]: 1b36f4b

File: tests/pkcli/build_data/1.out/srv/host/v9.radia.run/srv/rsiviz/start (hunk 5 above)
Comment (e-carlin) [reply tests/pkcli/build_data/1.out/srv/host/v9.radia.run/srv/rsiviz/start]: 1b36f4b

File: tests/pkcli/build_data/1.out/srv/host/v9.radia.run/srv/rsiviz/remove (hunk 4 above)
Comment (e-carlin) [reply tests/pkcli/build_data/1.out/srv/host/v9.radia.run/srv/rsiviz/remove]: 1b36f4b

File: tests/pkcli/build_data/1.out/srv/host/v9.radia.run/rsiviz.sh (hunk 3 above)
Comment (e-carlin) [reply tests/pkcli/build_data/1.out/srv/host/v9.radia.run/rsiviz.sh]: 1b36f4b

File: tests/pkcli/build_data/1.out/srv/host/v9.radia.run/srv/rsiviz/cmd (hunk 2 above)
Comment (e-carlin) [reply tests/pkcli/build_data/1.out/srv/host/v9.radia.run/srv/rsiviz/cmd]: 1b36f4b

File: rsconf/component/rsiviz.py (hunk 1 above)
Comment (e-carlin) [reply rsconf/component/rsiviz.py:68]: Possibly but as we add new modules to rsiviz and add configuration to them I think we'd have to update this. My hope when I originally added this block of code was we wouldn't have to change code as we add new configuration for rsiviz. Unless there is a regex that could cover it I'm not seeing?

Comment (robnagler) [reply rsconf/component/rsiviz.py:68]: I don't think there's an easy answer. It depends on knowledge of the whole system.

**Review by robnagler — APPROVED** [review]

**Pull request merged**